pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/json/ImmutableJsonIndexReader.java

```diff
@@ -396,7 +459,7 @@ private int[] getDictIdRangeForKey(String key) {
    *  Else, return the json path that is generated by replacing array index with . on the original key
    *  and the associated flattenDocId bitmap
    */
-  private Pair<String, MutableRoaringBitmap> getKeyAndFlattenDocId(String key) {
+  private Pair<String, MutableRoaringBitmap> getKeyAndFlattenedDocId(String key) {
```

Comment:
```suggestion
  private Pair<String, MutableRoaringBitmap> getKeyAndFlattenedDocIds(String key) {
```

Reply:
In the new syntax, I think we want to pass in the real json path, e.g. `$.foo`, or do we want to keep the current way of no dollar sign?